src/components/hooks/useDelayedSavedState.ts

```diff
@@ -14,10 +14,19 @@ export function useDelayedSavedState(
   formValue?: string,
   saveAfter?: number | boolean,
 ): DelayedSavedStateRetVal {
-  const [immediateState, setImmediateState] = React.useState(formValue);
+  const [immediateState, _setImmediateState] = React.useState(formValue);
+  const immediateStateRef = React.useRef(formValue);
   const [saveNextChangeImmediately, setSaveNextChangeImmediately] = React.useState(false);
   const [skipNextValidation, setSkipNextValidation] = React.useState(false);
 
+  const setImmediateState = React.useCallback(
+    (value: string | undefined) => {
+      immediateStateRef.current = value;
+      _setImmediateState(value);
```

Comment:
Could we avoid duplicating the state here and just use `immediateStateRef.current` every time? I'm not sure what the purpose of double-storing it is.

Reply:
Is it not necessary to use State for the hook to trigger an update?

Reply:
Oh, right, yep - that's right!